src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11258,6 +11258,7 @@ private ParameterSyntax ParseLambdaParameter(bool isFirst, ref bool hasTypes)
             var pk = this.PeekToken(1).Kind;
             if (isRefOrOutOrParams
                 || (pk != SyntaxKind.CommaToken && pk != SyntaxKind.CloseParenToken && (hasTypes || isFirst))
+                || (this.CurrentToken.Kind == SyntaxKind.OpenParenToken && pk == SyntaxKind.CloseParenToken && (hasTypes || isFirst))
```

Comment:
It looks like these two conditions (the new line and the one before it) both have a common part ```&& (hasTypes || isFirst)```. Consider merging them together so that this portion is evaluated only once. Also, consider adding a comment about the situation we are trying to detect.